mssql/base.py

```diff
@@ -571,6 +573,36 @@ def __init__(self, cursor, connection):
         self.last_sql = ''
         self.last_params = ()
 
+    def _as_sql_type(self, typ, value):
+        if value is None:
+            return 'INT'
```

Comment:
I think this will generate errors.
For example, if the column where I use this value is of type datetime, it will fail as seen in #340.
Maybe we have to replace None with nulls in our query?

Reply:
Maybe we can use sql_variant?

Reply:
I have not been able to generate this with the ORM, but using raw i was able to make it fail:

```python
query = Comment.objects.raw("SELECT * from testapp_comment where created_at = %s GROUP BY id", params=(None,))
print(list(query))
```

The generated query is:

```python
DECLARE @var0 INT = ?  
SELECT * from testapp_comment where created_at = @var0 GROUP BY id
```

And it fails:
`pyodbc.DataError: ('22018', '[22018] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Operand type clash: datetime2 is incompatible with int (206) (SQLExecDirectW)')`

It is a silly example, but I wonder what else we can break